InventoryManagementSystem_SQL/DataBaseOperations/InventoryService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Text;
+
+namespace InventoryManagementSystem
+{
+    public class InventoryService
+    {
+        private IProduct _repository;
+
+        public InventoryService(IProduct repository)
+        {
+            _repository = repository;
+        }
+
+        public async Task AddProductAsync(Product newProduct)
+        {
+            await _repository.AddProductAsync(newProduct);
+        }
+
+        public async Task<bool> IsEmptyAsync()
+        {
+            return await _repository.IsEmptyAsync();
+        }
+
+        public async Task<string> PrintAllProductsAsync()
+        {
+            if (await IsEmptyAsync())
+                return "There are no products.";
+
+            var allProducts = new StringBuilder();
+            var products = await _repository.GetAllProductsAsync();
+
+            for (var i = 0; i < products.Count; i++)
+            {
+                allProducts.Append($"""
+                Product #{i + 1}:
+                {products[i].ToString()}
+                
+
+                """);
+            }
+
+            return allProducts.ToString();
+        }
+
+        public async Task EditProductNameAsync(Product product, string newName)
+        {
+            await _repository.EditProductNameAsync(product.Name, newName);
+        }
+
+        public async Task EditProductPriceAsync(Product product, decimal newPrice)
+        {
+            await _repository.EditProductPriceAsync(product.Name, newPrice);
+        }
+
+        public async Task EditProductQuantityAsync(Product product, int newQuantity)
```

Comment:
no need to pass the whole product object if not used
the name can be sufficient